### DISCUSSION TITLE
Support for high level language

### DISCUSSION THREAD

**Discussion by Blquinn** [slint-ui/slint#2343] [Ideas] [answered]
Hello, I was wondering if slint has plans to support any statically typed, high level languages such as swift, C#, Java etc.

I think a lot of people, who want to use slint for desktop / mobile apps, will want to have a language that has reasonable performance and maintainability, but isn't a systems language like rust or c++, or a scripting language like JavaScript.

I was just wondering if you had given any thought to that, or have plans already.

**Comment by tronical** [comment, marked as answer]
Our design very much supports adding bindings to statically typed languages. We do have bindings for C++ and Rust that sort of prove that. The effort required to do this includes writing a code generator in our compiler to generate Swift/C#/Java and implementing a run-time library in the target language that can create data structures the Slint run-time library requires, as well as to call into the C based low-level API of the Slint run-time library.

We don't have any concrete/immediate plans to implement one of the above ourselves in the coming months though, but we're happy to guide anyone willing to work on it.

**Reply by Blquinn** [reply]
Ok, that's what I figured. Thanks for the info! Perhaps at some point I will have to give that a shot :)

**Reply by mateli** [reply]
A full C based API would be more useful as most high level languages has tools to take C headers and make wrappers. An example of this is Qt where most high level bindings use a simple C API. Also obviously C in itself is an important language.
